src/app/_components/content-navigation.tsx

```
@@ -0,0 +1,51 @@
+import Link from "next/link";
+
+type Props = {
+  content: string;
+};
+
+function extractHeadings(markdown: string): { level: number; text: string }[] {
+  const headingRegex = /^(#{1,6})\s+(.*)$/gm;
+  const headings = [];
+  let match: RegExpExecArray | null;
+
+  while ((match = headingRegex.exec(markdown)) !== null) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)